spring-cloud-starter-stream-source-sftp/src/test/java/org/springframework/cloud/stream/app/sftp/source/postprocessor/SftpContentTypeEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.sftp.source.postprocessor;
+
+import org.springframework.cloud.stream.app.postprocessor.ContentTypeEnvironmentPostProcessor;
+
+/**
+ * Sets the content type to use if not specified by user properties.
+ *
+ * @author Chris Schaefer
+ */
+public class SftpContentTypeEnvironmentPostProcessor extends ContentTypeEnvironmentPostProcessor {
```

Comment:
I found the problem that this class is in the `test` scope not `main`. Therefore it isn't visible and not includes into the target jar.

I guess you tested it with the default state and realized that `TaskLaunchRequest` is serialized and deserialized pretty easy with the default JSON content-type. Therefore from here I gues we don't need this specific `ContentTypeEnvironmentPostProcessor` and we don't care about `application/x-java-serialized-object` any more.

Can you elaborate, @chrisjs ?

Thanks

Reply:
no - my guess is i simply added the class to the test package by accident, my tests ran all nicely without JSON mapping and when i would pull in a snapshot of task-launcher 2.0.0 into DataFlow for testing things just magically aligned. 

i don't doubt serializing / deserializing objects is anything complicated. i just don't understand why one would want JSON by */default/* rather than a type, maybe I'm a dinosaur :^)

anyway.. PR is updated to remove this content type stuff. unfortunately now the tests have JSON conversion code in 🙉